src/m2k_impl.cpp

```diff
@@ -315,6 +316,134 @@ std::vector<M2kAnalogOut*> M2kImpl::getAllAnalogOut()
 	return m_instancesAnalogOut;
 }
 
+bool M2kImpl::hasMixedSignal()
+{
+	auto digital_impl = dynamic_cast<M2kDigitalImpl*>(getDigital());
+	return digital_impl->hasRateMux();
+}
+
+void M2kImpl::startMixedSignalAcquisition(unsigned int nb_samples)
+{
+	bool hasAnalogTrigger;
+	bool hasDigitalTrigger;
+	analogSource = m_trigger->getAnalogSource();
+	digitalSource = m_trigger->getDigitalSource();
+
+	hasAnalogTrigger = this->hasAnalogTrigger();
```

Comment:
separate declaration and assigning of value. You could do `bool ... = this->has...` on this line

Reply:
Modified.